### PR TITLE
Fix the testing strategy

### PR DESCRIPTION
**Pull request by zackkrida**
Fixes the`test` npm script and moves the `tests` directory to the repo root, outside of the main `src` directory. 

As discussed in the [preact-cli issues](https://github.com/developit/preact-cli/issues/555)  Follows @developit's preference of moving the `tests` directory.

### CODE REVIEW

File: template/package.json [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "setupFiles": [
       "<rootDir>/src/tests/__mocks__/browserMocks.js"
```
Comment (reznord) [inline template/package.json]: this should be:

```
<rootDir>/tests/__mocks__/browserMocks.js
```

since we have moved tests out of the `src` folder

**Comment by reznord** [comment]
In `package.json`, you forgot to update the [`moduleNameMapper`](https://github.com/zackkrida/default/blob/d5def277e679136628cc5ef15540cc434b54c97d/template/package.json#L49) config

**Review by reznord — CHANGES_REQUESTED** [review]

**Comment by reznord** [comment]
and you forgot to remove the `tests/` from the `src/`  folder 😅

**Comment by zackkrida** [comment]
@reznord yikes. my bad 😊

**Comment by reznord** [comment]
This is solved in #20, closing this.

**Pull request closed without merging**